android-pay/src/main/java/com/stripe/wrap/pay/utils/CartManager.java

```diff
@@ -114,46 +127,25 @@ public CartManager(@NonNull Cart oldCart, boolean shouldKeepShipping, boolean sh
      */
     @Nullable
     public String addLineItem(@NonNull @Size(min = 1) String description, long totalPrice) {
-        return addLineItem(description, totalPrice, LineItem.Role.REGULAR);
-    }
-
-    /**
-     * Adds a {@link LineItem.Role#REGULAR} item to the cart with a description
-     * and total price value. Currency matches the currency of the {@link CartManager}.
-     *
-     * @param description a line item description
-     * @param totalPrice the total price of the line item, in the smallest denomination
-     * @param role the {@link LineItem.Role} for this item
-     * @return a {@link String} UUID that can be used to access the item in this {@link CartManager}
-     * or {@code null} if this is a {@link LineItem.Role#TAX} item
-     */
-    @Nullable
-    public String addLineItem(
-            @NonNull @Size(min = 1) String description,
-            long totalPrice,
-            int role) {
```

Comment:
Turns out having two different (String, number, number) constructors that mean very different things is really bad. Now, for each type, we just have (String, number)  => "Description", "Total Price", and (String, number, number) => "Description", "quantity", "unit price". You differentiate between shipping and regular items by calling addLineItem and addShippingLineItem